### PR TITLE
DEM-933 Implement search result column sort

### PR DESCRIPTION
**Pull request by yordanalexandrov**
+ Added mat sort to table
+ Added configuration for sorting.
+ Added tests

### CODE REVIEW

File: projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+  /**
+   * The data source for the table
+   */
   @Input()
   public datasource: any[];
 
+  /**
+   * Table configuration, containing table and column specific configuration
+   * If column configuration is changed dynamically the whole columnConfigurations
+   * array must be overridden for changes to be detected
```
Comment (svilenvelikov) [inline projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: You mean that a new instance of the array should be created or what?

File: src/app/components/search-results/search-results.component.ts [hunk 2]
```diff
@@ -50,12 +54,35 @@ export class SearchResultsComponent implements OnInit {
           dataTemplate: this.template
         }
       ],
-      showFooter: true
+      showFooter: true,
+      enableSort: true,
+      defaultSortColumn: 'name',
+      defaultSortDir: 'asc'
```
Comment (svilenvelikov) [inline src/app/components/search-results/search-results.component.ts]: The two defaults above are optional parameters. Right? Where you set their default values?

File: projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts (hunk 1 above)
Comment (yordanalexandrov) [reply projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: Yes. But I will change that.
I am thinking about implementing DoCheck doing the diff checks manually.
This way there will be no need for that.

Comment (svilenvelikov) [reply projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: But why do you consider changing it? I kind of prefer the current approach.

Comment (yordanalexandrov) [reply projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: I think it is nicer for the component to take care of this as it provides the type/interface for the configuration.

Comment (svilenvelikov) [reply projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: ok

**Review by svilenvelikov — APPROVED** [review]

File: src/app/components/search-results/search-results.component.ts (hunk 2 above)
Comment (yordanalexandrov) [reply src/app/components/search-results/search-results.component.ts]: Those should not be in the config. I've now put them in a separate object passed trough an Input to the component.
They denote by which column and direction is the table currently sorted. This usually comes from the service response.

File: projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts (hunk 1 above)
Comment (yordanalexandrov) [reply projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts]: I have decided to leave it as is. I have clarified the comment.

**Review by teodossidossev — APPROVED** [review]

**Pull request merged**